storage/innobase/handler/ha_innodb.cc

```diff
@@ -15950,42 +16016,67 @@ static int innobase_alter_encrypt_tablespace(handlerton *hton, THD *thd,
       !(oldenc.empty() || Encryption::is_none(oldenc.data()));
   /* If new tablespace definition says it's encrypted */
   bool is_new_encrypted = !Encryption::is_none(newenc.data());
+  fil_space_crypt_t *crypt_data = space->crypt_data;
 
   // It is only possible to mark tablespace to be skipped by encryption threads
   // if it is not already encrypted.
   if (!is_old_encrypted && !is_new_encrypted &&
       alter_info->explicit_encryption) {
-    if (!fil_crypt_exclude_tablespace_from_rotation(space)) {
-      my_error(ER_EXCLUDE_ENCRYPTION_THREADS_RUNNING, MYF(0), space->name);
+    if (!fil_crypt_exclude_tablespace_from_rotation_permanently(space)) {
```

Comment:
Can you please add testcase to reach this code?

Reply:
There are couple of ENCRYPTION='N' set for unencrypted tables/tablespaces. Please have a look at encrypt_threads_skip_tablespace.test, line 121.